applicationLibrary/molecularDynamics/molecularDynamicsLibrary/LJFunctorHWY.h

```diff
@@ -0,0 +1,1405 @@
+    if constexpr (applyShift) {
+      shift6s = highway::Load(tag_double, shifts);
+    }
+  }
+
+  /**
+   * Actual inner kernel of the SoAFunctors
+   *
+   * @tparam newton3
+   * @tparam remainderI
+   * @tparam remainderJ
+   * @tparam reversed
+   * @tparam vecPattern
+   * @param i
+   * @param j
+   * @param ownedMaskI
+   * @param ownedStatePtr2
+   * @param x1
+   * @param y1
+   * @param z1
+   * @param x2Ptr
+   * @param y2Ptr
+   * @param z2Ptr
+   * @param fx2Ptr
+   * @param fy2Ptr
+   * @param fz2Ptr
+   * @param typeID1Ptr
+   * @param typeID2Ptr
+   * @param fxAcc
+   * @param fyAcc
+   * @param fzAcc
+   * @param virialSumX
+   * @param virialSumY
+   * @param virialSumZ
+   * @param uPotSum
+   * @param restI
+   * @param restJ
+   */
+  template <bool newton3, bool remainderI, bool remainderJ, bool reversed, VectorizationPattern vecPattern>
+  inline void SoAKernel(const size_t i, const size_t j, const MaskDouble &ownedMaskI,
+                        const int64_t *const __restrict ownedStatePtr2, const VectorDouble &x1, const VectorDouble &y1,
+                        const VectorDouble &z1, const double *const __restrict x2Ptr,
+                        const double *const __restrict y2Ptr, const double *const __restrict z2Ptr,
+                        double *const __restrict fx2Ptr, double *const __restrict fy2Ptr,
+                        double *const __restrict fz2Ptr, const size_t *const typeID1Ptr, const size_t *const typeID2Ptr,
+                        VectorDouble &fxAcc, VectorDouble &fyAcc, VectorDouble &fzAcc, VectorDouble &virialSumX,
+                        VectorDouble &virialSumY, VectorDouble &virialSumZ, VectorDouble &uPotSum,
+                        const unsigned int restI, const unsigned int restJ) {
+    VectorDouble epsilon24s = _epsilon24;
+    VectorDouble sigmaSquareds = _sigmaSquared;
+    VectorDouble shift6s = _shift6;
+
+    if constexpr (useMixing) {
+      fillPhysicsRegisters<remainderI, remainderJ, reversed, vecPattern>(typeID1Ptr, typeID2Ptr, epsilon24s,
+                                                                         sigmaSquareds, shift6s, j, restI, restJ);
+    }
+
+    VectorDouble x2;
+    VectorDouble y2;
+    VectorDouble z2;
+    VectorLong ownedStateJLong;
+
+    fillJRegisters<remainderJ, vecPattern>(j, x2Ptr, y2Ptr, z2Ptr, ownedStatePtr2, x2, y2, z2, ownedStateJLong, restJ);
+
+    // distance calculations
+    const auto drX = x1 - x2;
+    const auto drY = y1 - y2;
+    const auto drZ = z1 - z2;
+
+    const auto drX2 = drX * drX;
+    const auto drY2 = drY * drY;
+    const auto drZ2 = drZ * drZ;
+
+    const auto dr2 = drX2 + drY2 + drZ2;
+
+    const auto cutoffMask = highway::Le(dr2, _cutoffSquared);
+    const auto dummyMask =
+        highway::And(ownedMaskI, highway::Ne(highway::ConvertTo(tag_double, ownedStateJLong), _ownedStateDummy));
+    const auto cutoffDummyMask = highway::And(cutoffMask, dummyMask);
```

Comment:
```suggestion
    const auto dummyMask =
        highway::And(ownedMaskI, highway::MaskFromVec(highway::ConvertTo(tag_double, ownedStateJLong)));
    const auto cutoffDummyMask = highway::MaskedLe(dummyMask, dr2, _cutoffSquared);
```

I think this should do the same thing and be slightly better performing.